app/[locale]/(dashboard)/stations/[id]/edit/page.jsx

```diff
@@ -1,11 +1,32 @@
 import { getServerSession } from 'next-auth/next'
+import { redirect } from 'next/navigation'
 import { authOptions } from '@/lib/auth'
 import { Page } from '@/components/Page/Page'
 import { PageHeader } from '@/components/Page/PageHeader'
 import { BackpageButton } from '@/components/ui/Button/BackpageButton'
+import EditStationForm from '@/components/Form/EditStationForm'
+import { Edit } from '@/components/ui/Icons'
 
-async function Home({ params }) {
+const getStationById = async (id) => {
+  try {
+    const res = await fetch(`http://localhost:3000/api/stations/${id}`, {
+      cache: 'no-store',
+    })
+    if (!res.ok) throw new Error('Failed to fetch station')
+    const json = await res.json()
+    return json
+  } catch (error) {
+    console.log(error)
+  }
+}
+
+export default async function EditStation({ params }) {
   const session = await getServerSession(authOptions)
+  const { id } = params
+  const { station } = await getStationById(id)
+  const { name, type, sensors, longitude, latitude, description, image_url } =
```

Comment:
I don't know if opening the station variable like this is a good practice... maybe passing the hole station to the component is better, and inside the component we decide which attributes we will get or not